### PR TITLE
Fix sdk client's createDirectus JSdoc

### LINKED ISSUE #23960: Fix SDK's createDirectus JSDoc inaccuracy
### Page

https://github.com/directus/directus/blob/e9c1cbd3353375a26586a168075cb679443da1b1/sdk/src/client.ts#L17

### What's wrong?
The function `createDirectus` accepts two parameters: `url` and `options`.
The JSDoc annotations above it  says it accepts `url` and `config`.
  
Among other uses, JSDoc annotations are used in IDE and Code Editors to inform users on what to pass the correct parameters to functions. An outdated/wrong annotation might induce users to make errors.
 
### Where?
[github permalink](https://github.com/directus/directus/blob/e9c1cbd3353375a26586a168075cb679443da1b1/sdk/src/client.ts#L17)

### How to fix

- [ ] Change `config` to `options` in the JSDoc
- [ ] [Optional] Explain what options are possible or use `@see` to link the `ClientOptions` interface

### PR DESCRIPTION
**Pull request by ztickm**
## Scope

This PR only touches documentation.

What's changed:

- Added types to the two @params and to the @return
- Renamed config @param to options
- Marked options @param as optional using square brackets
- Linked accepted and related types for options @param 

## Review Notes / Questions

- If it exists, I would like to know if there's another place where this is documented 

---

Fixes #23960

### CODE REVIEW

File: sdk/src/client.ts [hunk 1]
```diff
@@ -13,10 +13,11 @@ const defaultGlobals: ClientGlobals = {
 /**
  * Creates a client to communicate with a Directus app.
  *
- * @param url The URL to the Directus app.
- * @param config The optional configuration.
+ * @param {string} url - The URL to the Directus app.
+ * @param {ClientOptions} [options] - The client options. Defaults to defaultGlobals.
+ *     {@see {@link ClientOptions}} and {@see {@link ClientGlobals}}
```
Comment (paescuj) [inline sdk/src/client.ts:18]: We don't need types in JSDoc as they are already provided by TypeScript.

```suggestion
 * @param url The URL to the Directus app.
 * @param options The client options. Defaults to the standard implementation of `globals`.
```

File: sdk/src/client.ts [hunk 2]
```diff
@@ -13,10 +13,11 @@ const defaultGlobals: ClientGlobals = {
...
  *
- * @returns A Directus client.
+ * @returns {DirectusClient} A Directus client.
```
Comment (paescuj) [inline sdk/src/client.ts:20]:
```suggestion
 * @returns A Directus client.
```

**Review by paescuj — APPROVED** [review]
Thanks @ztickm!

**Comment by ztickm** [comment]
It's a pleasure !

**Pull request merged**